Upstream 5.x dropped src/caffe/opencv-caffe.proto and only regenerates the
tensorflow/onnx protos with PROTOBUF_UPDATE_FILES, so the dnn build cannot
find opencv-caffe.pb.h when using the system protobuf. Restore the proto
(copied from the 4.x branch into ${FILESDIR}) and regenerate it too.

--- opencv/modules/dnn/CMakeLists.txt.orig	2025-07-04 16:37:46 UTC
+++ opencv/modules/dnn/CMakeLists.txt
@@ -393,7 +393,7 @@
   ocv_target_compile_definitions(${the_module} PRIVATE "HAVE_PROTOBUF=1")
 
   if(PROTOBUF_UPDATE_FILES)
-    file(GLOB proto_files "${CMAKE_CURRENT_LIST_DIR}/src/tensorflow/*.proto" "${CMAKE_CURRENT_LIST_DIR}/src/onnx/opencv-onnx.proto")
+    file(GLOB proto_files "${CMAKE_CURRENT_LIST_DIR}/src/tensorflow/*.proto" "${CMAKE_CURRENT_LIST_DIR}/src/caffe/opencv-caffe.proto" "${CMAKE_CURRENT_LIST_DIR}/src/onnx/opencv-onnx.proto")
     if(CMAKE_VERSION VERSION_LESS "3.13.0")
       set(PROTOBUF_GENERATE_CPP_APPEND_PATH ON) # required for tensorflow
       protobuf_generate_cpp(fw_srcs fw_hdrs ${proto_files})
